### ISSUE TITLE
Output plots using downloaded raster from bounds2raster doesn't plot correctly

### ISSUE THREAD

**Issue by JackMcKew** [geopandas/contextily#155]
I was recently trying to use a downloaded raster as the background map for some plots with geopandas. However noticed that some of the plots were just offset from the actual location. This problem only seems to occur when using downloaded rasters.

I initially thought this might be an extents problem, however both plots are showing the exact same extents.

I've included a reproducible example straight from the geopandas docs: <https://geopandas.readthedocs.io/en/latest/gallery/plotting_basemap_background.html?highlight=basemap>

```python
import geopandas
import contextily

# Example from https://geopandas.readthedocs.io/en/latest/gallery/plotting_basemap_background.html?highlight=basemap

df = geopandas.read_file(geopandas.datasets.get_path('nybb'))

# Project to web mercator
df = df.to_crs(epsg=3857)

# Get boundaries of dataset
west, south, east, north = bbox = df.total_bounds

# Plot geodataframe
ax = df.plot(figsize=(10, 10), alpha=0.5, edgecolor='k')

# Add basemap through contextily
contextily.add_basemap(ax,source=contextily.providers.CartoDB.Positron)

# Show limits are the same
print(ax.get_xlim(),ax.get_ylim())
```

    (-8269187.064680938, -8201155.139990268) (4935220.954753395, 5002970.199586581)
    
![contextily-bug_0_1](https://user-images.githubusercontent.com/19178331/83250598-ba1bec00-a1eb-11ea-928f-d2deb7687df2.png)

```python
import geopandas
import contextily

# Example from https://geopandas.readthedocs.io/en/latest/gallery/plotting_basemap_background.html?highlight=basemap

df = geopandas.read_file(geopandas.datasets.get_path('nybb'))

# Project to web mercator
df = df.to_crs(epsg=3857)

# Get boundaries of dataset
west, south, east, north = bbox = df.total_bounds

# Get raster 
img, ext = contextily.bounds2raster(west,
                             south,
                             east,
                             north,
                             "test.tif",
                             source=contextily.providers.CartoDB.Positron)

# Plot geodataframe
ax = df.plot(figsize=(10, 10), alpha=0.5, edgecolor='k')

# Add basemap from downloaded raster
contextily.add_basemap(ax,source='test.tif')

# Show limits are the same
print(ax.get_xlim(),ax.get_ylim())
```

    (-8269187.064680938, -8201155.139990268) (4935220.954753395, 5002970.199586581)
    
![contextily-bug_1_1](https://user-images.githubusercontent.com/19178331/83250613-c2742700-a1eb-11ea-9b97-d8a266c36f4a.png)

**Comment by darribas** [comment]
Thanks very much for the report!

Mmm... It's a strange one. I've played a bit and , if you set `reset_extent=False`, it aligns both correctly. Now in this setup, I think that should not matter, and in any case the behaviour you highlight should not happen anyway...

Will investigate!

**Comment by darribas** [comment]
OK I've done some exploration and I think I've narrowed it to this point:

https://github.com/geopandas/contextily/blob/master/contextily/plotting.py#L191

We set the `extent` for the rendering for the whole file whether you `reset_extent` or not. We need to catch it if `reset_extent=True` and adapt it to the extent of the `window` instead. PRs welcome or I'll be trying to issue a patch soon too.

**Comment by darribas** [comment]
OK, I've issued a PR to fix it over at #156

**Comment by JackMcKew** [comment]
That was so fast! Thank you so much.

After running into this problem for a day or so, I finally realized that contextily caches the data anyway which is what I was trying to do by storing the local file! https://contextily.readthedocs.io/en/latest/intro_guide.html#Store-basemaps-locally

> First, in a given session, each tile is only downloaded once. Tiles are stored in a temporary cache so the next time you need it, you don’t have go to the web. This results in important time savings and network footprint.

This package is awesome, thank you so much for your work! 👍
